### PR TITLE
refactor: const char* context_json to string_view

### PR DESCRIPTION
**Pull request by olgavrou**
- Replace `const char* context_json` with `string_view context_json` 
- Updated rl.net.live_model.cc and cs bindings to reflect this change
- Updated python bindings to reflect this change
- changed `rl_string_view.h` to include and reuse vw's string view implementation

Next step could be to do the same change for event_id

### CODE REVIEW

File: rlclientlib/multistep.cc [hunk 1]
```diff
@@ -2,12 +2,13 @@
...
 namespace reinforcement_learning {
-  void episode_history::update(const char* event_id, const char* previous_event_id, const char* context, const ranking_response& resp) {
+  void episode_history::update(const char* event_id, const char* previous_event_id, string_view context, const ranking_response& resp) {
     _depths[event_id] = this->get_depth(previous_event_id) + 1;
   }
 
-  std::string episode_history::get_context(const char* previous_event_id, const char* context) const {
-    return R"({"episode":{"depth":")" + std::to_string(this->get_depth(previous_event_id) + 1) + "\"}," + std::string(context + 1);
+  std::string episode_history::get_context(const char* previous_event_id, string_view context) const {
+    // is the +1 here important?
```
Comment (olgavrou) [inline rlclientlib/multistep.cc:10]: @ataymano

File: include/rl_string_view.h [hunk 2]
```diff
@@ -0,0 +1,9 @@
+#pragma once
+
+// Rely on the VW implementation of string_view
+#include "../ext_libs/vowpal_wabbit/vowpalwabbit/vw_string_view.h"
```
Comment (jackgerrits) [inline include/rl_string_view.h:4]: A public header cant have a local repo dependent include path like this. This wont work when installed. 

It should be either `vowpalwabbit/vw_string_view.h` or `vw_string_view.h` with the appropriate include path made available via the VowpalWabbit::vw dependency

File: rlclientlib/ranking_event.h [hunk 3]
```diff
@@ -61,11 +62,11 @@ namespace reinforcement_learning {
     learning_mode get_learning_mode() const;
 
   public:
-    static ranking_event choose_rank(const char* event_id, const char* context,
+    static ranking_event choose_rank(const char* event_id, string_view context,
```
Comment (jackgerrits) [inline rlclientlib/ranking_event.h:65]: Should all strings be string views?

File: rlclientlib/ranking_event.h [hunk 4]
```diff
@@ -123,11 +124,11 @@ namespace reinforcement_learning {
     const std::string& get_event_id() const;
 
   public:
-    static multi_slot_decision_event request_decision(const std::string& event_id, const char* context,
+    static multi_slot_decision_event request_decision(const std::string& event_id, string_view context,
```
Comment (jackgerrits) [inline rlclientlib/ranking_event.h:127]: Should event_id be a string_view?

Comment (olgavrou) [reply rlclientlib/ranking_event.h:127]: see comment in description :)

File: include/rl_string_view.h (hunk 2 above)
Comment (olgavrou) [reply include/rl_string_view.h:4]: we should change future_compat.h then too? how does it work with that?

File: rlclientlib/ranking_event.h (hunk 4 above)
Comment (jackgerrits) [reply rlclientlib/ranking_event.h:127]: Ah I see. I vote for event_id being done as well, otherwise it feels inconsistent.

Comment (olgavrou) [reply rlclientlib/ranking_event.h:127]: yeah I agree but it's too much for one PR I think

File: include/rl_string_view.h (hunk 2 above)
Comment (jackgerrits) [reply include/rl_string_view.h:4]: Yep, we need to fix that one too.

Comment (olgavrou) [reply include/rl_string_view.h:4]: OK I can fix both together in a separate PR too, how does that sound?

Comment (jackgerrits) [reply include/rl_string_view.h:4]: Sounds good!

File: bindings/cs/rl.net.native/rl.net.native.vcxproj [hunk 5]
```diff
@@ -57,7 +57,7 @@
       <Optimization>Disabled</Optimization>
       <PreprocessorDefinitions>_DEBUG;_CONSOLE;%(PreprocessorDefinitions)</PreprocessorDefinitions>
       <SDLCheck>false</SDLCheck>
-      <AdditionalIncludeDirectories>$(ProjectDir)..\..\..\include;%(AdditionalIncludeDirectories)</AdditionalIncludeDirectories>
+      <AdditionalIncludeDirectories>$(ProjectDir)..\..\..\include;$(SolutionDir)\ext_libs\vowpal_wabbit\vowpalwabbit;$(SolutionDir)\ext_libs\vowpal_wabbit\explore;%(AdditionalIncludeDirectories)</AdditionalIncludeDirectories>
```
Comment (lokitoth) [inline bindings/cs/rl.net.native/rl.net.native.vcxproj:60]: Is there a corresponding change that needs to be made to the CMakeLists.txt file here?

**Review by lokitoth — APPROVED** [review]

File: bindings/cs/rl.net.native/rl.net.native.vcxproj (hunk 5 above)
Comment (jackgerrits) [reply bindings/cs/rl.net.native/rl.net.native.vcxproj:60]: These transitive includes are automatically added by CMake. For some reason msbuild doesnt want to propagate include directories from included projects

**Pull request merged**